spec/models/ucas_matched_application_spec.rb

```diff
@@ -25,6 +25,24 @@
 
       expect(ucas_matching_application.course).to eq(course)
     end
+
+    it 'returns the course details for a course which is not on Apply' do
+      ucas_matching_data =
+        { 'Scheme' => 'U',
+          'Course code' => '123',
+          'Course name' => 'Not on Apply',
+          'Provider code' => course.provider.code.to_s,
+          'Apply candidate ID' => candidate.id.to_s }
+      ucas_matching_application = UCASMatchedApplication.new(ucas_matching_data, recruitment_cycle_year)
+
+      course_struct = OpenStruct.new(
+        code: '123',
+        name: 'Not on Apply',
+        provider: Provider.find_by(code: course.provider.code.to_s),
+      )
+
+      expect(ucas_matching_application.course).to eq(course_struct)
```

Comment:
Rather than checking equality for the whole struct, perhaps assert that the `course` exposes the right interface, eg `course.name == 'Not on apply'`?